### DISCUSSION TITLE
How do I grab single records from a JSON file?

### DISCUSSION THREAD

**Discussion by dannyvalle** [nuxt/nuxt#18096] [Questions]
I am using Nuxt 3 – total newbie here – to setup a products section in a website that has categories. The JSON data file lives in /data/hardware.json:
`[
    {
        "category" : "category_name",
        "products" : [
            {
                "product_slug" : "item-1",
                "product_name" : "Item 1",
                "product_description" : "Lorem ipsum dolor sit amet consectetur adipisicing elit. Repellendus quae maxime."
            },
            {
                "product_slug" : "item-2",
                "product_name" : "Item 2",
                "product_description" : "Lorem ipsum dolor sit amet consectetur adipisicing elit. Repellendus quae maxime."
            }
        ]
    },
    {
        "category" : "category_name",
        "products" : [
            {
                "product_slug" : "item-1",
                "product_name" : "Item 1",
                "product_description" : "Lorem ipsum dolor sit amet consectetur adipisicing elit. Repellendus quae maxime."
            },
            {
                "product_slug" : "item-2",
                "product_name" : "Item 2",
                "product_description" : "Lorem ipsum dolor sit amet consectetur adipisicing elit. Repellendus quae maxime."
            }
        ]
    }
]`

I am using the server/api folder to grab the data in my file located here: /server/api/hardware.ts

`import data from "~/data/hardware.json";
export default (req, res) => {
	return data;
};`

In my pages directory, I have a [slug].vue file. How do I grab a single product item on that page? Are there any code demos I can reference to learn from?

**Comment by viii47** [comment]
Hey @dannyvalle,

Here's an example where the server-side/api will return a status and the dataset if it has been found. Be aware that there many approaches, e.g. just dumping the whole dataset to the frontend and verifying it there, using a `<v-for>` class to check it inside the template etc. or whichever makes you feel fancy.

### Template -> [slug].vue
```js
<template>
    <div class="container mx-auto">
        <span v-if="data.found">
            Product found :D
            {{ data.product.product_name }}
        </span>
        <span v-else>
            Product not found :(
        </span>
    </div>
</template>

<script lang="ts">
export default defineComponent({
    async setup() {
        const route = useRoute();
	// See https://v3.nuxtjs.org/docs/directory-structure/server
	// Assumption: Server API created in ../server/api/hardware/info.ts
        const { data } = await useFetch(`/api/hardware/info`,{
            body: {
                slug: route.params.slug
            }
        })
        return {
            data
        }
    }
})
</script>
``` 

### Server-API -> info.ts (../server/api/hardware/info.ts)
```js

import { useBody } from 'h3'
// Temporary workaround (see https://github.com/nuxt/framework/discussions/1247)
import destr from 'destr'

// Your JSON File 
// You can of course import it via `import data from "~/data/hardware.json"`
const testjson = [
   {
      "category":"category_name",
      "products":[
         {
            "product_slug":"item-1",
            "product_name":"Item 1",
            "product_description":"Lorem ipsum dolor sit amet consectetur adipisicing elit. Repellendus quae maxime."
         },
         {
            "product_slug":"item-2",
            "product_name":"Item 2",
            "product_description":"Lorem ipsum dolor sit amet consectetur adipisicing elit. Repellendus quae maxime."
         }
      ]
   },
   {
      "category":"category_name",
      "products":[
         {
            "product_slug":"item-1",
            "product_name":"Item 1",
            "product_description":"Lorem ipsum dolor sit amet consectetur adipisicing elit. Repellendus quae maxime."
         },
         {
            "product_slug":"item-2",
            "product_name":"Item 2",
            "product_description":"Lorem ipsum dolor sit amet consectetur adipisicing elit. Repellendus quae maxime."
         }
      ]
   }
];

export default async (req, res) => {
  const body =  req.body ? destr(req.body) : await useBody(req)
  // Setting up the default response
  let data = {
    found: false,
    product: {}
  }
  // Using Array.prototype.forEach to find the right dataset 
  // We're using the product_slug property as the key to find the right dataset 
  // Note: I'm solely going through the first index to demonstrate it
  // If you have multiple children then you need to react according (e.g. transmit multiple params/routes)
  // You could also just dump the whole dataset to the frontend and verify it there
  // Flavor-yyy is up to you. You can use reduce, filter, Object.keys or whatever your heart desires.
  testjson[0].products.forEach((productset: any, index: any) => {
    if (productset.product_slug.toLowerCase() === body.slug.toLowerCase()) {
      data.found = true;
      data.product = productset;
    }
  });
  return data
}
```
